p2p/dial.go

```diff
@@ -196,35 +202,81 @@ func (s *dialstate) taskDone(t task, now time.Time) {
 		s.hist.add(t.dest.ID, now.Add(dialHistoryExpiration))
 		delete(s.dialing, t.dest.ID)
 	case *discoverTask:
-		if t.bootstrap {
-			s.bootstrapped = true
-		}
 		s.lookupRunning = false
 		s.lookupBuf = append(s.lookupBuf, t.results...)
 	}
 }
 
 func (t *dialTask) Do(srv *Server) {
-	addr := &net.TCPAddr{IP: t.dest.IP, Port: int(t.dest.TCP)}
-	glog.V(logger.Debug).Infof("dialing %v\n", t.dest)
+	if t.dest.Incomplete() {
+		if !t.resolve(srv) {
+			return
+		}
+	}
+	success := t.dial(srv, t.dest)
+	// Try resolving the ID of static nodes if dialing failed. If we
+	// already resolved them earlier (probably because the config
+	// holds an incomplete node), don't do it again.
+	if !success && t.flags&staticDialedConn != 0 {
```

Comment:
Edit: Are you referring to the check in `resolve` which basically enforces once per minute maximum? Or is there another larger scope check that ensures static nodes are resolved once per the lifetime of the process?